cpp/src/aztec3/circuits/rollup/base/native_base_rollup_circuit.cpp

```diff
@@ -360,7 +365,7 @@ AppendOnlySnapshot check_nullifier_tree_non_membership_and_insert_to_tree(BaseRo
                         }
                     }
                     // if not matched, our subtree will misformed - we must reject
-                    assert(matched);
+                    composer.do_assert(matched, "Nullifier subtree is malformed");
                     _unused(matched);
```

Comment:
We can remove _unused() function calls (and maybe also the `_unused` function itself)